js-pkg/packages/driftdb-react/src/index.tsx

```diff
@@ -65,7 +65,7 @@ export function DriftDBProvider(props: {
     return () => {
       dbRef.current?.disconnect()
     }
-  }, [])
+  }, [props.room, props.useBinary, props.api, dbRef.current])
```

Comment:
I don't think it's recommended to put refs in dependency arrays since they can lead to some unpredictable behavior. They don't cause rerenders when they change, so the only way this useEffect will get run again with a new dbRef value (were it to change), would be if another value in the props array _incidentally_ changed. Also, in this case the dbRef.current value is always accessed from dbRef itself and never passed as a value to another function, so there's no need to worry about it being stale when this effect runs.

There's a paragraph in [this article](https://epicreact.dev/the-latest-ref-pattern-in-react/) - about halfway down - that talks a little about this.